internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -505,10 +505,26 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	if err := CheckRegionConfigsPriorityOrder(params.GetReplicationSpecs()); err != nil {
 		return diag.FromErr(err)
 	}
-	// cannot call latest API (2024-10-23 or newer) as it can enable ISS autoscaling
-	cluster, _, err := connV220240805.ClustersApi.CreateCluster(ctx, projectID, params).Execute()
-	if err != nil {
-		return diag.FromErr(fmt.Errorf(errorCreate, err))
+
+	var clusterName string
+	var clusterID string
+	var err error
+	if isUsingOldShardingConfiguration(d) {
```

Comment:
Include old comment? // cannot call latest API (2024-10-23 or newer) as it can enable ISS autoscaling

Reply:
good point, added clarifying comments